### ISSUE TITLE
[google_maps_flutter] Update Marker's position without rebuilding the whole Map 

### ISSUE THREAD

**Issue by Melak12** [flutter/flutter#103130]
Hello, 
I am working on a ride-hailing project using Flutter. I want to show the driver's movement on the map as smooth as uber. However, I can't with the google_maps_flutter package. This is due to the whole Google Map object is rebuilt when there is a position change in the car marker. Is there any way to improve this?

**Comment by ZhuBoao** [comment]
Hi I am not from flutter team and I happened to see your issue on top. I thinks your goal can be achieved by build your marker inside a `Stack` on top of Google Map and calculate the offset on the screen manually. Google map uses Web Mercator projection, so you can build your calculation base on that. I can share my implementation with you.
```dart
extension LatLngExtension on LatLng {
  // get geodetic latitude in radians
  double getPhi() {
    return latitude * pi / 180;
  }

  // get longitude in radians
  double getLambda() {
    return longitude * pi / 180;
  }

  // get X coordinate values after Web Mercator
  double getX(double zoom) {
    return ((256 / 2 / pi) * pow(2, zoom) * (this.getLambda() + pi));
  }

  // get X coordinate values after Web Mercator
  double getY(double zoom) {
    return ((256 / 2 / pi) *
        pow(2, zoom) *
        (pi - log(tan(pi / 4 + this.getPhi() / 2))));
  }

  double distanceInKm(LatLng point) {
    double lat1 = point.latitude;
    double lat2 = this.latitude;
    double lon1 = point.longitude;
    double lon2 = this.longitude;

    var R = 6371; // Radius of the earth in km
    var dLat = _deg2rad(lat2 - lat1); // deg2rad below
    var dLon = _deg2rad(lon2 - lon1);
    var a = sin(dLat / 2) * sin(dLat / 2) +
        cos(_deg2rad(lat1)) *
            cos(_deg2rad(lat2)) *
            sin(dLon / 2) *
            sin(dLon / 2);
    var c = 2 * atan2(sqrt(a), sqrt(1 - a));
    var d = R * c; // Distance in km
    return d;
  }

  _deg2rad(deg) {
    return deg * (pi / 180);
  }

  String toLngLatString() {
    return "$longitude,$latitude";
  }
}

class LocationHelper {
  static double getLatDistance(LatLng a, LatLng b, double zoom) {
    if (a.latitude == b.latitude) {
      return 0;
    }
    return b.getY(zoom) - a.getY(zoom);
  }

  static double getLngDistance(LatLng a, LatLng b, double zoom) {
    if (a.longitude == b.longitude) {
      return 0;
    }
    if ((a.longitude - b.longitude).abs() > 180) {
      return LatLng(a.latitude, -a.longitude).getX(zoom) - b.getX(zoom);
    }
    return b.getX(zoom) - a.getX(zoom);
  }

  static double getDistance(LatLng a, LatLng b, double zoom) {
    return sqrt(
        pow(a.getX(zoom) - b.getX(zoom), 2) + pow(a.getY(1) - b.getY(1), 2));
  }

  static Offset latLngToOffset(
      LatLng latLng, LatLng cameraTarget, double cameraZoom) {
    double xOffset =
        LocationHelper.getLngDistance(cameraTarget, latLng, cameraZoom);
    double yOffset =
        LocationHelper.getLatDistance(cameraTarget, latLng, cameraZoom);
    Offset result = Offset(xOffset, yOffset);
    return result;
  }
}
```

In your widget you can do :
```dart
 Offset offset =  LocationHelper.latLngToOffset(widget.currentLocation, widget.cameraTarget, widget.cameraZoom);
```
to get the offset

**Comment by Melak12** [comment]
Thanks for the prompt response. Your way sounds very legit. Let me try with that.

**Comment by neoacevedo** [comment]
I saw this similar behaviour with  another plugin: **flutter_typeahead**, but hapens being stopped in a place and typing on flutter_typeahead's textfield widget.

On it, I use such plugin for addresses autocomplete. We tested the app and was working good, we have published the app in the play store and it is in production, but today, installing from the play store, having it published without changes, the map again is redrawing totally:
https://photos.app.goo.gl/ag5ma1yAnXAet78z8 

Once redrawed, it raises the issue described at https://github.com/flutter/flutter/issues/43785 and it is constant when happens, but is solved if I exit from the app and open the app again. I think, the context is glitchy at first run but is fixed when the app is reopened.
